synapse/python_dependencies.py

```diff
@@ -76,7 +76,7 @@
     "netaddr>=0.7.18",
     "Jinja2>=2.9",
     "bleach>=1.4.3",
-    "typing-extensions>=3.7.4",
+    "typing-extensions>=3.10.0",
```

Comment:
Any issues with bumping this?

Reply:
I'll add a comment.

Reply:
> Any issues with bumping this?

Not sure I understand the question. What should I be looking out for?

Reply:
From the package maintainers: "Fedora 34 and 35 has 3.7.4.3. Only Fedora 36 will get 3.10.0.2".
Fedora 34 and 35 EOL is 17th May 2022 and 7th December 2022 respectively.

So we either have to wait until December to merge this, or see if the packagers are willing to bring 3.10.0.2 to Fedora 34/35.